### ISSUE TITLE
Rule proposal: disallow curlies around string literals

### ISSUE THREAD

**Issue by lencioni** [yannickcr/eslint-plugin-react#529]
Bad:

``` jsx
function MyComponent() {
  return (
    <div className={'MyComponent'}>
      Hi mom
    </div>
  )
}
```

Good: 

``` jsx
function MyComponent() {
  return (
    <div className='MyComponent'>
      Hi mom
    </div>
  )
}
```

**Comment by yannickcr** [comment]
So, the opposite of [jsx-no-literals](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-literals.md) ? If yes we could make a `jsx-literals` rule with `always`/`never` options.

**Comment by lencioni** [comment]
Yes, I think so. For the `never` case, there are probably some exceptions when used outside of props (e.g. `{' '}` is often used).

**Comment by ljharb** [comment]
I'd think this is `jsx-string-literals-in-props` or something - I don't think we'd want this rule to legislate text used outside of props.

**Comment by lencioni** [comment]
Yeah, that's fair. I could see folks wanting different things in props vs outside props, so having separate rules seems like a decent way to achieve that.

**Comment by yannickcr** [comment]
Mmm, apparently the current `jsx-no-literals` rule only apply outside props.

**Comment by ljharb** [comment]
Seems closed by #1349.
